ArmPkg/Library/ArmLib/AArch64/AArch64Support.masm

```diff
@@ -1,679 +1,758 @@
-    ands  x0, x0, #AARCH64_PFR0_FP// Extract bits indicating VFP implementation
-    cmp   x0, #0                  // VFP is implemented if '0'.
-    bne   %f4                      // Exit if VFP not implemented.
-    // FVP is implemented.
-    // Make sure VFP exceptions are not trapped (to any exception level).
-    mrs   x0, cpacr_el1           // Read EL1 Coprocessor Access Control Register (CPACR)
-    orr   x0, x0, #CPACR_VFP_BITS // Disable FVP traps to EL1
-    msr   cpacr_el1, x0           // Write back EL1 Coprocessor Access Control Register (CPACR)
-    mov   x1, #AARCH64_CPTR_TFP   // TFP Bit for trapping VFP Exceptions
-    EL1_OR_EL2_OR_EL3(x2)
-1
-    ret                           // Not configurable in EL1
-2
-    mrs   x0, cptr_el2            // Disable VFP traps to EL2
-    bic   x0, x0, x1
-    msr   cptr_el2, x0
-    ret
-3
-    mrs   x0, cptr_el3            // Disable VFP traps to EL3
-    bic   x0, x0, x1
-    msr   cptr_el3, x0
-4
-    ret
-ArmEnableVFP ENDP
-
-
-ArmCallWFI PROC
-  wfi
-  ret
-ArmCallWFI ENDP
-
-
-ArmReadMpidr PROC
-  mrs   x0, mpidr_el1           // read EL1 MPIDR
-  ret
-ArmReadMpidr ENDP
-
-
-// Keep old function names for C compatibilty for now. Change later?
-ArmReadTpidrurw PROC
-  mrs   x0, tpidr_el0           // read tpidr_el0 (v7 TPIDRURW) -> (v8 TPIDR_EL0)
-  ret
-ArmReadTpidrurw ENDP
-
-
-// Keep old function names for C compatibilty for now. Change later?
-ArmWriteTpidrurw PROC
-  msr   tpidr_el0, x0           // write tpidr_el0 (v7 TPIDRURW) -> (v8 TPIDR_EL0)
-  ret
-ArmWriteTpidrurw ENDP
-
-
-// Arch timers are mandatory on AArch64
-ArmIsArchTimerImplemented PROC
-  mov   x0, #1
-  ret
-ArmIsArchTimerImplemented ENDP
-
-
-ArmReadIdPfr0 PROC
-  mrs   x0, id_aa64pfr0_el1   // Read ID_AA64PFR0 Register
-  ret
-ArmReadIdPfr0 ENDP
-
-
-// Q: id_aa64pfr1_el1 not defined yet. What does this funtion want to access?
-// A: used to setup arch timer. Check if we have security extensions, permissions to set stuff.
-//    See: ArmPkg/Library/ArmArchTimerLib/AArch64/ArmArchTimerLib.c
-//    Not defined yet, but stick in here for now, should read all zeros.
-ArmReadIdPfr1 PROC
-  mrs   x0, id_aa64pfr1_el1   // Read ID_PFR1 Register
-  ret
-ArmReadIdPfr1 ENDP
-
-// VOID ArmWriteHcr(UINTN Hcr)
-ArmWriteHcr PROC
-  msr   hcr_el2, x0        // Write the passed HCR value
-  ret
-ArmWriteHcr ENDP
-
-// UINTN ArmReadHcr(VOID)
-ArmReadHcr PROC
-  mrs   x0, hcr_el2
-  ret
-ArmReadHcr ENDP
-
-// UINTN ArmReadCurrentEL(VOID)
-ArmReadCurrentEL PROC
-  mrs   x0, CurrentEL
-  ret
-ArmReadCurrentEL ENDP
-
-// UINT32 ArmReadCntHctl(VOID)
-ArmReadCntHctl PROC
-  mrs   x0, cnthctl_el2
-  ret
-ArmReadCntHctl ENDP
-
-// VOID ArmWriteCntHctl(UINT32 CntHctl)
-ArmWriteCntHctl PROC
-  msr   cnthctl_el2, x0
-  ret
-ArmWriteCntHctl ENDP
-
-
-    END
-
+//------------------------------------------------------------------------------
+//
+// Copyright (c) 2008 - 2010, Apple Inc. All rights reserved.<BR>

```

Comment:
Line endings? Is the whole commit fixing up other commits?

Reply:
Looks like the original commit did add it as LF. Updated all commits touching the file to CRLF.